### PR TITLE
Add Bookshelf Attraction

### PR DESCRIPTION
**Pull request by erasels**
Adds an attraction and Framework to render text. Right now NUM_PAGES is set to 1 until enough Pages are added.

Example:
image

### CODE REVIEW

File: src/main/java/spireCafe/interactables/attractions/bookshelf/BookshelfAttraction.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                    filteredPages.add(p);
+                }
+            }
+        }
+        if(pages.size() < NUM_PAGES) {
+            pages.addAll(filteredPages, 1); // Reduced so that the last few pages have a higher chance to be seen
+        }
+
+        for (int i = 0; i < NUM_PAGES; i++) {
+            AbstractPage p = pages.getRandom(AbstractDungeon.miscRng, true);
```
Comment (modargo) [inline src/main/java/spireCafe/interactables/attractions/bookshelf/BookshelfAttraction.java:75]: Now that I think about it... once `NUM_PAGES` is more than 1, won't this be able to get the same page more than once? Is that intended? I think we'd want to remove the selected page from the list.

File: src/main/java/spireCafe/interactables/attractions/bookshelf/BookshelfAttraction.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                } else {
+                    filteredPages.add(p);
+                }
+            }
+        }
+        if(pages.size() < NUM_PAGES) {
+            pages.addAll(filteredPages, 1); // Reduced so that the last few pages have a higher chance to be seen
+        }
+
+        for (int i = 0; i < NUM_PAGES; i++) {
```
Comment (modargo) [inline src/main/java/spireCafe/interactables/attractions/bookshelf/BookshelfAttraction.java:74]: One other thought that occurred to me: yYou could set `NUM_PAGES` to a higher value right now, and change this line to `for (int i = 0; i < Math.min(NUM_PAGES, pages.size(); i++) {`.

File: src/main/java/spireCafe/interactables/attractions/bookshelf/BookshelfAttraction.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        }
+    }
+
+    @Override
+    public void renderCutscenePortrait(SpriteBatch sb) {}
+
+    @Override
+    public void onInteract() {
+        List<AbstractCafeInteractable> inhabitants = ((CafeRoom) AbstractDungeon.getCurrRoom().event).getCurrentInhabitants();
+        List<String> seenPages = getSeenPages();
```
[the remark this answers is not in the document]
Comment (erasels) [reply src/main/java/spireCafe/interactables/attractions/bookshelf/BookshelfAttraction.java:57]: canSpawn is always checked as the filteredPages are also assigned only if the page canSpawn. Preferred spawn is a good point though

File: src/main/java/spireCafe/interactables/attractions/bookshelf/BookshelfAttraction.java (hunk 1 above)
Comment (erasels) [reply src/main/java/spireCafe/interactables/attractions/bookshelf/BookshelfAttraction.java:75]: The second input param being true means the item will be removed

File: src/main/java/spireCafe/interactables/attractions/bookshelf/BookshelfAttraction.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (modargo) [reply src/main/java/spireCafe/interactables/attractions/bookshelf/BookshelfAttraction.java:57]: Makes sense with canSpawn, I overlooked seenPages and filteredPages being different. That's good, means that if you want to change this the preferredSpawn is the only thing that needs adjusting.

File: src/main/java/spireCafe/interactables/attractions/bookshelf/BookshelfAttraction.java (hunk 1 above)
Comment (modargo) [reply src/main/java/spireCafe/interactables/attractions/bookshelf/BookshelfAttraction.java:75]: Sounds good, that's what I get for not looking closely enough at the Weighted List implementation :)

**Pull request merged**